server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -154,6 +159,33 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             }
         }
 
+        @Override
+        public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper, QueryShardContext context) {
+            if (hasDocValues()) {
```

Comment:
we don't need this `if` block, do we? All 6.x and 7.x indices have a single type.

Reply:
Don't we still support multi-type indices that have been created in 5.x on 6.x? I will remove this for the PR against master but I think we need it for 6.x in that case. Can you confirm?

Reply:
:+1: